src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -157,16 +162,19 @@ private async Task AcceptClientsAsync()
                     if (error != null)
                         throw error;
 
-                    if (!this.AllowClientConnection(tcpClient))
+                    (bool successful, string reason) connectionAttempt = this.AllowClientConnection(tcpClient);
+                    if (!connectionAttempt.successful)
                     {
+                        this.signals.Publish(new PeerConnectionAttemptFailed(true, (IPEndPoint)tcpClient.Client.RemoteEndPoint, connectionAttempt.reason));
```

Comment:
Is this perhaps also needed inside the outer `catch` block catching `Exception`? I suppose there isn't much more that can fail after this point in the code.

Reply:
mmm I'm not sure, the only way to go in that catch is if this.AllowClientConnection throws or L163, and anyway in that case the NetworkPeerServer AcceptClientsAsync will exit, meaning the server would stop listening for incoming peers...  I don't know why we have chosen to not catch inside the loop and maybe in case of that exception (that 
shouldn't happen) just wait few second before continuing to listen for connection

Who knows, maybe that error sometimes throw and we stop without knowing, probably we should write that in bold... or better use AsyncProvider, could be worthy to open an issue maybe